src/box/lua/schema.lua

```diff
@@ -2075,6 +2075,20 @@ base_index_mt.fselect = function(index, key, opts, fselect_opts)
         num_cols = math.max(num_cols, #tab[i])
     end
 
+    -- The JSON encoder above passes through binary data unmodified.
+    -- Use Base64 encoding for such strings.
+    local digest = require('digest')
+    for j = 1,num_cols do
+        for i = 1,num_rows do
+            if tab[i][j] then
+                local _, err = utf8.len(tab[i][j])
```

Comment:
What happens if binary data comprises a valid, but non-printable UTF8 string? Maybe, we should mask such strings, too, as we do in Console/YaML:

https://github.com/tarantool/tarantool/blob/5557c47a39f36c1d00291050854fa9cbf1382ae1/third_party/lua-yaml/lyaml.cc#L667

Reply:
Valid non-printable chars are escaped by `json.encode` above.